internal/hook/hook.go

```diff
@@ -438,26 +439,46 @@ func (ha *Argument) Get(r *Request) (string, error) {
 	case SourceHeader:
 		source = &r.Headers
 		key = textproto.CanonicalMIMEHeaderKey(ha.Name)
+
 	case SourceQuery, SourceQueryAlias:
 		source = &r.Query
+
 	case SourcePayload:
 		source = &r.Payload
+
 	case SourceString:
 		return ha.Name, nil
+
+	case SourceRequest:
+		if r == nil || r.RawRequest == nil {
+			return "", errors.New("request is nil")
+		}
+
+		switch ha.Name {
```

Comment:
Let's do `strings.ToLower(ha.Name)` so that we avoid issues with "Method" not being matched :-)

Reply:
Fixed